atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1900,6 +1900,17 @@ private void putMetadataWithoutChangingSettings(final TableReference tableRef, f
         delete(AtlasDbConstants.DEFAULT_METADATA_TABLE, oldVersions);
     }
 
+    @Override
+    public void deleteRange(TableReference tableRef, RangeRequest range) {
+        // TODO: implement an actual range delete using the cassandra client
+        // As a cheap optimization, use truncate when the range is for the whole table.
+        if (range.equals(RangeRequest.all())) {
+            truncateTable(tableRef);
```

Comment:
this would be much safer as
try {truncate table}
catch (YourClusterDoesntHaveEveryoneUpToDoASchemaChangeRightNow e) {
   retry as actual deleteRange
}